app/services/catalog/copy_portfolio_item.rb

```diff
@@ -0,0 +1,29 @@
+module Catalog
+  class CopyPortfolioItem
+    attr_reader :new_portfolio_item
+
+    def initialize(params)
+      @portfolio_item = PortfolioItem.find(params[:portfolio_item_id])
+      @to_portfolio = Portfolio.find(params[:portfolio_id] || @portfolio_item.portfolio_id)
+    end
+
+    def process
+      @new_portfolio_item = make_copy
+      @to_portfolio.portfolio_items << @new_portfolio_item
+
+      self
+    end
+
+    private
+
+    def make_copy
+      @portfolio_item.dup.tap do |new_portfolio_item|
+        if new_portfolio_item.portfolio_id == @to_portfolio.id
+          new_portfolio_item.name = "Copy of " + new_portfolio_item.name
```

Comment:
@lindgrenj6 if the copy is called multiple times, they would all be called Copy of NGINIX, Copy of NGINX, Should this have a counter that we see in Windows Explorer Copy(1) of NGINIX, Copy(2) of NGINIX etc.
Also how about asking the user to provide you with a new name, without a provided name it uses the default Copy (#) of NGINX